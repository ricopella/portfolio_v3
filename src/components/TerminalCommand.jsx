import React from 'react'
import styled from '@emotion/styled'
import { colors } from '../styles/variables'

const TerminalWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: max-content max-content;
  margin: 2rem 0;
`

const TerminalInput = styled.div`
  color: ${colors.gray.calm};
`

const TerminalOutput = styled.div`
  color: ${colors.accent};
`

const TerminalArrow = styled.span`
  color: ${colors.limeGreen};
`
const TerminalLink = styled.a`
  cursor: pointer;
  color: ${colors.teal};
`

const FinalStatement = styled.div`
  content: '';
  display: inline-block;
  background-color: ${colors.code};
  vertical-align: top;
  width: 10px;
  height: 24px;
  -webkit-animation: blink 1s step-end infinite;
  animation: blink 1s step-end infinite;

  @-webkit-keyframes blink {
    0% {
      opacity: 1;
    }
    50% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }

  @keyframes blink {
    0% {
      opacity: 1;
    }
    50% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
`

const TerminalCommand = ({ question, answer, isLink }) => (
  <TerminalWrapper>
    {question && answer ? (
      <>
        <TerminalInput>
          <TerminalArrow>>></TerminalArrow> {question}
        </TerminalInput>
        <TerminalOutput>
          {!isLink ? (
            answer
          ) : (
            <TerminalLink
              href={isLink}
              target="_blank"
              rel="noopener noreferrer"
            >
              "{answer}"
            </TerminalLink>
          )}
        </TerminalOutput>
      </>
    ) : (
      <>
        <TerminalInput>
          <TerminalArrow>>></TerminalArrow> <FinalStatement />
        </TerminalInput>
      </>
    )}
  </TerminalWrapper>
)

export default TerminalCommand
