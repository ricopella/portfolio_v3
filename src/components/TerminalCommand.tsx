import React, { FC } from 'react'
import styled from '@emotion/styled'
import { colors } from '../styles/variables'
import { TerminalCommandProps } from '../types'

const TerminalWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: max-content max-content;
  margin: 2rem 0;
`

const TerminalInput = styled.div`
  color: ${colors.gray.calm};
`

const TerminalOutput = styled.div`
  color: ${colors.accent};
`

const TerminalArrow = styled.span`
  color: ${colors.limeGreen};
`
const TerminalLink = styled.a`
  cursor: pointer;
  color: ${colors.teal};
`

const FinalStatement = styled.div`
  content: '';
  display: inline-block;
  background-color: ${colors.code};
  vertical-align: top;
  width: 0.625rem;
  height: 1.5rem;
  -webkit-animation: blink 1s step-end infinite;
  animation: blink 1s step-end infinite;

  @-webkit-keyframes blink {
    0% {
      opacity: 1;
    }
    50% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }

  @keyframes blink {
    0% {
      opacity: 1;
    }
    50% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
`

const TerminalCommand: FC<TerminalCommandProps> = ({
  question,
  answer,
  href,
}) => (
  <TerminalWrapper>
    {question && answer ? (
      <>
        <TerminalInput>
          <TerminalArrow>>></TerminalArrow> {question}
        </TerminalInput>
        <TerminalOutput>
          {!href ? (
            answer
          ) : (
            <TerminalLink href={href} target="_blank" rel="noopener noreferrer">
              "{answer}"
            </TerminalLink>
          )}
        </TerminalOutput>
      </>
    ) : (
      <>
        <TerminalInput>
          <TerminalArrow>>></TerminalArrow> <FinalStatement />
        </TerminalInput>
      </>
    )}
  </TerminalWrapper>
)

export default TerminalCommand
