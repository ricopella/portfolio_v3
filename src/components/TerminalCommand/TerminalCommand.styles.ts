import styled from '@emotion/styled'
import { COLORS } from '../../styles/variables'

const TerminalWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: max-content max-content;
  margin: 2rem 0;
`

const TerminalInput = styled.div`
  color: ${COLORS.gray.calm};
`

const TerminalOutput = styled.div`
  color: ${COLORS.accent};
`

const TerminalArrow = styled.span`
  color: ${COLORS.limeGreen};
`
const TerminalLink = styled.a`
  cursor: pointer;
  color: ${COLORS.teal};
`

const FinalStatement = styled.div`
  content: '';
  display: inline-block;
  background-color: ${COLORS.code};
  vertical-align: top;
  width: 0.625rem;
  height: 1.5rem;
  -webkit-animation: blink 1s step-end infinite;
  animation: blink 1s step-end infinite;

  @-webkit-keyframes blink {
    0% {
      opacity: 1;
    }
    50% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }

  @keyframes blink {
    0% {
      opacity: 1;
    }
    50% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
`

export default {
  FinalStatement,
  TerminalLink,
  TerminalArrow,
  TerminalOutput,
  TerminalInput,
  TerminalWrapper,
}
