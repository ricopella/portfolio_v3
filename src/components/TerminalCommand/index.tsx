import React, { FC } from 'react'
import { TerminalCommandProps } from '../../types'
import Styled from './TerminalCommand.styles'

const TerminalCommand: FC<TerminalCommandProps> = ({
  question,
  answer,
  href,
}) => (
  <Styled.TerminalWrapper>
    {question && answer ? (
      <>
        <Styled.TerminalInput>
          <Styled.TerminalArrow>>></Styled.TerminalArrow> {question}
        </Styled.TerminalInput>
        <Styled.TerminalOutput>
          {!href ? (
            answer
          ) : (
            <Styled.TerminalLink
              href={href}
              target="_blank"
              rel="noopener noreferrer"
            >
              "{answer}"
            </Styled.TerminalLink>
          )}
        </Styled.TerminalOutput>
      </>
    ) : (
      <>
        <Styled.TerminalInput>
          <Styled.TerminalArrow>>></Styled.TerminalArrow>{' '}
          <Styled.FinalStatement />
        </Styled.TerminalInput>
      </>
    )}
  </Styled.TerminalWrapper>
)

export default TerminalCommand
