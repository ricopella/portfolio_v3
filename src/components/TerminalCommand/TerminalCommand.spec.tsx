import renderer from 'react-test-renderer'
import React from 'react'
import TerminalCommand from './index'
import { render } from '@testing-library/react'

describe('<TerminalCommand />', () => {
  const props = {
    question: `Who?`,
    answer: `Wu-Tang Clan`,
    href: `wutang.com`,
  }

  it('should match TerminalCommand snapshot', () => {
    const internalTree = renderer
      .create(<TerminalCommand {...props} />)
      .toJSON()

    expect(internalTree).toMatchSnapshot()
  })

  it('should render question and answer', () => {
    const { getByText } = render(<TerminalCommand {...props} />)

    expect(getByText(/Wu-Tang Clan/i)).toBeTruthy()
    expect(getByText(/Who?/i)).toBeTruthy()
  })
})
