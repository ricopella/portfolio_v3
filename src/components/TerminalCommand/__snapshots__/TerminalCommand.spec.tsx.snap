// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TerminalCommand /> should match TerminalCommand snapshot 1`] = `
<div
  className="css-3yu43t"
>
  <div
    className="css-ntk5lm"
  >
    <span
      className="css-zbjqpz"
    >
      &gt;&gt; 
    </span>
     Who?
  </div>
  <div
    className="css-1yzmtjz"
  >
    <a
      className="css-1t2r17y"
      href="wutang.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      "Wu-Tang Clan"
    </a>
  </div>
</div>
`;
